### PR TITLE
Help: Rethink the way we render the HelpContactForm

### PR DESCRIPTION
**Pull request by omarjackman**
As previously discussed with @dmsnell I modified the logic that sets up the rendering for the `HelpContactForm`. Instead of using multiple functions that return `HelpContactForm` react classes I now have a single point where the React class is used and I'm using functions to provide the appropriate properties

### CODE REVIEW

**Comment by omarjackman** [comment]
cc @dmsnell

File: client/me/help/help-contact/index.jsx [hunk 1]
```diff
@@ -228,10 +222,15 @@ module.exports = React.createClass( {
...
 		if ( olark.isOperatorAvailable ) {
-			return this.getChatForm();
+			contactFormProps = this.getChatFormProperties();
+		} else {
+			contactFormProps = this.getKayakoFormProperties();
 		}
 
-		return this.getKayakoTicketForm();
+		// Hide the olark widget in the bottom right corner.
+		olarkActions.hideBox();
```
Comment (dmsnell) [inline client/me/help/help-contact/index.jsx]: This is just a DOM manipulation for a global element, right? Just checking that we aren't attempting to change the component's state mid-render.

Comment (omarjackman) [inline client/me/help/help-contact/index.jsx]: > This is just a DOM manipulation for a global element, right? Just checking that we aren't attempting to change the component's state mid-render.

Thats correct.

**Comment by dmsnell** [comment]
Looks good @omarjackman. The gains here aren't as big as I had hoped. On the other hand, I can see how the early-return logic in `getView()` is making this considerably easier. One final observation is that several of the props in the Kayako and Chat forms are the same, meaning we might be able to eliminate those two `get...Properties` calls with something of a hybrid...

**Comment by omarjackman** [comment]
@dmsnell I did a rebase which caused the logic here to get a bit more complicated. Take a look at the new changes.

**Comment by dmsnell** [comment]
Looks fine. Go for it!

**Pull request merged**